pkg/pool-weighted/contracts/smart/IndexPool.sol

```diff
@@ -93,11 +92,43 @@ contract IndexPool is BaseWeightedPool {
         // Immutable variables cannot be initialized inside an if statement, so we must do conditional assignments
         _tokens = tokens;
 
-        for(uint i = 0; i < numTokens; i++){
+        for (uint8 i = 0; i < numTokens; i++) {
             scalingFactors.push(_computeScalingFactor(tokens[i]));
         }
     }
 
+    function reweighTokens(address[] memory tokens, uint96[] memory normalizedWeights) public {
+        uint256 numTokens = tokens.length;
+        InputHelpers.ensureInputLengthMatch(numTokens, normalizedWeights.length);
+
+        uint256 normalizedSum = 0;
+        for (uint8 i = 0; i < numTokens; i++) {
+            uint256 normalizedWeight = normalizedWeights[i];
+
+            normalizedSum = normalizedSum.add(normalizedWeight);
```

Comment:
Can the weight of a token be zero as well? I know for the `reindexTokens()` function it can but not sure for this, and if not, maybe add a require. Or can we also use this function to get the weight to zero and then remove it with the `reindexTokens` function?

Reply:
I thought about this as well and I'm not yet entirely sure about it. It depends on the logic for removing tokens. Right now I believe it should be possible to set the weight to zero, even though that contracdicts a bulletpoint in the requirements.